src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -154,6 +148,31 @@ export function addItem(
   });
 }
 
+function replaceEquippedItem(
+  defs: D2ManifestDefinitions | D1ManifestDefinitions,
+  item: DimItem,
+  equip: boolean
+) {
+  return produce((draftLoadout) => {
+    setEquipForItemInLoadout(defs, item, draftLoadout, equip);
+  });
+}
+
+export function dropItem(
+  defs: D2ManifestDefinitions | D1ManifestDefinitions,
+  item: DimItem,
+  equip?: boolean,
+  socketOverrides?: SocketOverrides
+): LoadoutUpdateFunction {
+  const updateFunctions: LoadoutUpdateFunction[] = [];
+  if (item.bucket.hash === BucketHashes.Subclass) {
+    updateFunctions.push(clearSubclass(defs));
+  }
+  updateFunctions.push(replaceEquippedItem(defs, item, Boolean(equip)));
+  updateFunctions.push(addItem(defs, item, equip, socketOverrides));
+  return compose(...updateFunctions);
```

Comment:
Would this be easier to read if it were written like this? I'm not sure.

```suggestion
  return (loadout) => {
    if (item.bucket.hash === BucketHashes.Subclass) {
      loadout = clearSubclass(defs)(loadout);
    }
    loadout = replaceEquippedItem(defs, item, Boolean(equip))(loadout);
    loadout = addItem(defs, item, equip, socketOverrides)(loadout);
    return loadout;
  };
```

Reply:
Yeah I think it would, I think I have already convinced myself that we don't need the composition thing but I am not super keen on the `action(...args)(loadout)` that happens when you use actions directly. I might only partially do this as the `addItem` action is the big one I was avoiding splitting up in this PR.

Reply:
Ypu could always make an "applyUpdate" function to make it less weird, though I don't really mind this sort of thing.